connection.go

```diff
@@ -531,23 +533,38 @@ func (conn *Connection) dial() (err error) {
 
 func pack(h *smallWBuf, enc *encoder, reqid uint32,
 	req Request, streamId uint64, res SchemaResolver) (err error) {
+	const uint32Code = 0xce
+	const uint64Code = 0xcf
+	const streamBytesLenUint64 = 10
+	const streamBytesLenUint32 = 6
+
 	hl := h.Len()
 
+	var streamBytesLen = 0
+	var streamBytes [streamBytesLenUint64]byte
 	hMapLen := byte(0x82) // 2 element map.
 	if streamId != ignoreStreamId {
 		hMapLen = byte(0x83) // 3 element map.
+		streamBytes[0] = KeyStreamId
+		if streamId > math.MaxUint32 {
```

Comment:
Isn't there some `msgpack.dump` that could encode a single number? As far as I understand, any valid msgpack integer could be used here.

Reply:
We build the header manually, without using msgpack directly. Maybe it's a little faster (I didn't benchmark it myself).

To me, refactoring of the approach looks like out of scope the bug (we need benchamrks for it at least). I'd rather leave it as it is here at now.

Reply:
To be honest, I have similar issue in Python unpacking myself: there is some manual parsing of default structures that possibly could be processed by `msgpack` package. Unfortunately, sometimes Python `msgpack` module is not enough or really inconvenient to use, but maybe golang one is a bit better.

I still think that this approach is more prone to errors (we're basically reimplementing uint msgpack encoding here, and we're definitely cover less cases than module), but let it be like it for now.